src/aave-v2/SupplyVaultBase.sol

```diff
@@ -67,22 +85,28 @@ abstract contract SupplyVaultBase is ERC4626UpgradeableSafe, OwnableUpgradeable
         onlyInitializing
         returns (ERC20 underlyingToken)
     {
-        if (_poolToken == address(0)) revert ZeroAddress();
-
         poolToken = _poolToken;
 
-        underlyingToken = ERC20(IAToken(_poolToken).UNDERLYING_ASSET_ADDRESS());
+        underlyingToken = ERC20(IAToken(_poolToken).UNDERLYING_ASSET_ADDRESS()); // Reverts on zero address so no check for pool token needed.
         underlyingToken.safeApprove(address(morpho), type(uint256).max);
     }
 
     /// EXTERNAL ///
 
-    function transferTokens(
-        address _asset,
-        address _to,
-        uint256 _amount
-    ) external onlyOwner {
-        ERC20(_asset).safeTransfer(_to, _amount);
+    /// @notice Sets the rewards recipient.
+    /// @param _recipient The new rewards recipient.
+    function setRewardsRecipient(address _recipient) external onlyOwner {
+        recipient = _recipient;
+        emit RewardsRecipientSet(_recipient);
+    }
+
+    /// @notice Transfers the MORPHO rewards to the rewards recipient.
+    /// @dev Anybody can trigger this function. This offloads the DAO to do it.
```

Comment:
Quite tricky. If you change your rewardsDistributor for example (quite unlikely), you need to atomically update this contract.

Reply:
Yes, good point! But I think in general the process will be the following:
- deploy new `RewardsDistributor`
- update the recipient on vaults
- claim all rewards entitled to the vaults
- send them to the new `RewardsDistributor`

If all rewards entitled to the vaults are claimed quickly after Merkle's tree of core protocol's update, I don't think this is necessary for it to be atomic. Knowing, there's an incentive economic for vault's users to claim rewards quickly after update on behalf of the vault and send them to the `RewardsDistributor`.

Reply:
Even not atomically, if we decide for some reason to change the rewards distributor, there should be no token going to the previous one from this moment (which is impossible to ensure).

Otherwise the script creating the roots must depend on the last transfer, that we didn’t controlled. The simplest workflow or me is: when we want a new tree, we call the transfer function. Then we compute the tree, and push it.

For me it is more issues than issues solved.

Reply:
Why it is impossible to ensure? If there's no token to claim, then there is no issue no?

Reply:
Because the vaults are earning rewards at each block.

Reply:
But the drop of Morpho rewards is done only after the end of each epoch. This is continuous in theory but not in practice.

Reply:
Can you explain why?

Reply:
### Solution with only the DAO

You need to worry less in case of a rewardsDistributor change. You can do whatever you prefer, depending on the constraints, and your trees.

### Solution where everybody can transferRewards

You need to be careful in case of a rewardsDistributor change. Your trees need to depend on the last transfer that has been done to the old tree, which is not something you control.

Reply:
So if the probability of a rewards distributor change is 0, then the actual solution is better. Otherwise I don’t think so.

Reply:
What's the "but need to do the 90% rest of the job"? What you mentioned above?

To me, it's close to 0 yes we can ask @PaulFrambot or @julien-devatom about that

Reply:
The remaining 90% of the job is: computing the new merkle tree, and push the root on chain.

Reply:
> This a bundle of 13 txs to check each month which could be prevented by doing one single tx setting the correct rewards distributor.

Anyway, being realistic, the DAO will have to do these txs because it's extremely unlikely to me that someone else will transfer rewards to the rewards distributor
And at the same time, it's not much of a burden for the DAO to atomically update the recipient

So it's not an easy decision, but I'd say that keeping this privilege to the DAO is safer (if we ever can't figure out all the ins & outs of the problem) and is not much of a burden to the DAO

Reply:
But you can just use an EOA for doing so then.

A middle ground is to add the `onlyOwner` modifier so that the DAO still needs to trigger the function and there is no parameter to check?

Reply:
I don't think I'm relevant in this discussion as I don't make nor sign txs for the DAO

Reply:
>A middle ground is to add the onlyOwner modifier so that the DAO still needs to trigger the function and there is no parameter to check?

Yes that's perfect. Let's keep things simple (and it is simpler with an onlyOwner).